sdc/native/utils.cpp

```diff
@@ -26,4 +26,19 @@ void* upper_bound(void* first, void* last, void* value, int size, int item_size,
     return first;
 }
 
-}
\ No newline at end of file
+tbb::task_arena& get_arena()
+{
+    static tbb::task_arena arena;
+
+    if (!arena.is_active())
+        arena.initialize();
+
+    return arena;
+}
+
+void set_threads_num(uint64_t threads)
+{
+    get_arena().initialize(threads);
```

Comment:
I'm not sure it is going to work as intended. IMO Better to just save value to global variable here and than pass this var into arena ctor.
`static tbb::task_arena arena(num_threads);`